# - Replace the OpenFOAM third-party libraries with EASYBUILD variables.
# - Set install dir, compiler, etc using EASYBUILD
# <Mark.Olesen@esi-group.com>
--- OpenFOAM-v1706.orig/applications/utilities/mesh/manipulation/setSet/Allwmake	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/applications/utilities/mesh/manipulation/setSet/Allwmake	2017-07-05 13:36:32.826823741 +0200
@@ -9,11 +9,11 @@
 #
 # use readline if available
 #
-if [ -f /usr/include/readline/readline.h ]
+if [ -f $EBROOTLIBREADLINE/include/readline/readline.h ]
 then
     echo "    found <readline/readline.h>  --  enabling readline support."
     export COMP_FLAGS="-DHAS_READLINE"
-    export LINK_FLAGS="-lreadline"
+    export LINK_FLAGS="-L$EBROOTLIBREADLINE/lib -lreadline -L$EBROOTNCURSES -lncurses"
 fi
 
 wmake $targetType
--- OpenFOAM-v1706.orig/src/parallel/decompose/Allwmake	2017-06-27 15:04:26.213523686 +0200
+++ OpenFOAM-v1706/src/parallel/decompose/Allwmake	2017-07-11 11:42:41.467097531 +0200
@@ -39,7 +39,9 @@
 
     # Library
     [ -r $FOAM_EXT_LIBBIN/libmetis.so ] || \
+    [ -r $METIS_ARCH_PATH/lib/libmetis.a ] || \
     [ -r $METIS_ARCH_PATH/lib/libmetis.so ] || \
+    [ -r $METIS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libmetis.a ] || \
     [ -r $METIS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libmetis.so ] || \
     [ "${METIS_ARCH_PATH##*-}" = system ] || {
         echo "$warning (missing library)"
@@ -94,7 +96,9 @@
 
     # Library
     [ -r $FOAM_EXT_LIBBIN/libscotch.so ] || \
+    [ -r $SCOTCH_ARCH_PATH/lib/libscotch.a ] || \
     [ -r $SCOTCH_ARCH_PATH/lib/libscotch.so ] || \
+    [ -r $SCOTCH_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libscotch.a ] || \
     [ -r $SCOTCH_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libscotch.so ] || \
     [ "${SCOTCH_ARCH_PATH##*-}" = system ] || {
         echo "$warning (missing library)"
--- OpenFOAM-v1706.orig/src/parallel/decompose/ptscotchDecomp/Make/options	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/src/parallel/decompose/ptscotchDecomp/Make/options	2017-07-05 12:23:10.494399931 +0200
@@ -5,7 +5,6 @@
     $(PFLAGS) $(PINC) \
     -I$(SCOTCH_ARCH_PATH)/include/$(FOAM_MPI) \
     -I$(SCOTCH_ARCH_PATH)/include \
-    -I/usr/include/scotch \
     -I../decompositionMethods/lnInclude
 
 /*
--- OpenFOAM-v1706.orig/src/parallel/decompose/scotchDecomp/Make/options	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/src/parallel/decompose/scotchDecomp/Make/options	2017-07-05 12:24:29.590283915 +0200
@@ -8,7 +8,6 @@
 EXE_INC = \
     $(PFLAGS) $(PINC) \
     -I$(SCOTCH_ARCH_PATH)/include \
-    -I/usr/include/scotch \
     -I../decompositionMethods/lnInclude
 
 /*
--- OpenFOAM-v1706.orig/etc/bashrc	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/bashrc	2017-07-11 09:17:28.319350909 +0200
@@ -43,9 +43,10 @@
 # - If this does not produce the desired result, please set one of the fallback
 #   values to an appropriate path.
 #
-rc="${BASH_SOURCE:-${ZSH_NAME:+$0}}"
-[ -n "$rc" ] && FOAM_INST_DIR=$(\cd $(dirname $rc)/../.. && \pwd -L) || \
-FOAM_INST_DIR=$HOME/$WM_PROJECT
+##EasyBuild## module sets FOAM_INST_DIR
+# rc="${BASH_SOURCE:-${ZSH_NAME:+$0}}"
+# [ -n "$rc" ] && FOAM_INST_DIR=$(\cd $(dirname $rc)/../.. && \pwd -L) || \
+# FOAM_INST_DIR=$HOME/$WM_PROJECT
 # FOAM_INST_DIR=/opt/$WM_PROJECT
 # FOAM_INST_DIR=/usr/local/$WM_PROJECT
 #
@@ -90,7 +91,7 @@
 #- MPI implementation:
 #    WM_MPLIB = SYSTEMOPENMPI | OPENMPI | SYSTEMMPI | MPICH | MPICH-GM | HPMPI
 #               | MPI | FJMPI | QSMPI | SGIMPI | INTELMPI | USERMPI
-export WM_MPLIB=SYSTEMOPENMPI
+export WM_MPLIB=EASYBUILDMPI
 
 #- Operating System:
 #    WM_OSTYPE = POSIX
@@ -119,7 +120,7 @@
 
 # Location of third-party software
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-export WM_THIRD_PARTY_DIR=$WM_PROJECT_INST_DIR/ThirdParty-$WM_PROJECT_VERSION
+export WM_THIRD_PARTY_DIR=$WM_PROJECT_DIR/ThirdParty
 
 # Location of site-specific templates etc
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
--- OpenFOAM-v1706.orig/etc/config.sh/CGAL	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/CGAL	2017-07-05 12:43:36.924551892 +0200
@@ -50,41 +50,9 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_64_0
-cgal_version=CGAL-4.9.1
-
-export BOOST_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
-export CGAL_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
+##EasyBuild##
+export BOOST_ARCH_PATH=$EBROOTBOOST
+export CGAL_ARCH_PATH=$EBROOTCGAL
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ "$FOAM_VERBOSE" -a "$PS1" ]
-then
-    echo "Using boost ($boost_version)  ->  $BOOST_ARCH_PATH" 1>&2
-    echo "Using CGAL ($cgal_version)  ->  $CGAL_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLib > /dev/null 2>&1  # normal sourcing
-then
-
-    # If BOOST_ARCH_PATH, CGAL_ARCH_PATH do not end with '-system' or '-none',
-    # they are either located within ThirdParty, or a central installation
-    # outside of ThirdParty and must be added to the lib-path.
-
-    ending="${BOOST_ARCH_PATH##*-}"
-    if [ "$ending" != none -a "$ending" != system ]
-    then
-        _foamAddLib $BOOST_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-    fi
-
-    ending="${CGAL_ARCH_PATH##*-}"
-    if [ "$ending" != none -a "$ending" != system ]
-    then
-        _foamAddLib $CGAL_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-    fi
-
-    unset boost_version cgal_version ending
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/FFTW	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/FFTW	2017-07-05 12:51:03.067896983 +0200
@@ -48,31 +48,7 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-fftw_version=fftw-3.3.6-pl1
-export FFTW_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$fftw_version
+export FFTW_ARCH_PATH=$EBROOTFFTW
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ "$FOAM_VERBOSE" -a "$PS1" ]
-then
-    echo "Using fftw ($fftw_version)  ->  $FFTW_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLib > /dev/null 2>&1  # normal sourcing
-then
-
-    # If FFTW_ARCH_PATH does not end with '-system' or '-none',
-    # it is either located within ThirdParty, or a central installation
-    # outside of ThirdParty and must be added to the lib-path.
-
-    ending="${FFTW_ARCH_PATH##*-}"
-    if [ "$ending" != none -a "$ending" != system ]
-    then
-        _foamAddLib $FFTW_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-    fi
-
-    unset fftw_version ending
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/ensight	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/ensight	2017-07-05 12:51:46.179836814 +0200
@@ -29,26 +29,8 @@
 #     Sourced from OpenFOAM-*/etc/bashrc
 #
 #------------------------------------------------------------------------------
+##Easybuild## settings
 
-# Fallback value
-[ -d "$CEI_HOME" ] || export CEI_HOME=/usr/local/ensight/CEI
-
-if [ -r $CEI_HOME ]
-then
-
-    # Special treatment for 32bit OpenFOAM and 64bit Ensight
-    [ "$WM_ARCH-$(uname -m)" = linux-x86_64 ] && export CEI_ARCH=linux_2.6_32
-
-    # Add to path if required
-    if [ "$CEI_HOME/bin/ensight" != "$(command -v ensight)" ]
-    then
-        export PATH=$CEI_HOME/bin:$PATH
-    fi
-
-    export ENSIGHT9_INPUT=dummy
-    export ENSIGHT9_READER=$FOAM_LIBBIN
-else
-    unset CEI_HOME
-fi
+# unused
 
 #------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/gperftools	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/gperftools	2017-07-05 12:44:28.604471925 +0200
@@ -48,32 +48,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-gperftools_version=gperftools-2.5
-GPERFTOOLS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$gperftools_version
+GPERFTOOLS_VERSION=gperftools-$EBVERSIONGPERFTOOLS
+GPERFTOOLS_ARCH_PATH=$EBROOTGPERFTOOLS
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ "$FOAM_VERBOSE" -a "$PS1" ]
-then
-    echo "Using gperftools ($gperftools_version)  ->  $GPERFTOOLS_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLib > /dev/null 2>&1  # normal sourcing
-then
-
-    # If GPERFTOOLS_ARCH_PATH does not end with '-system' or '-none',
-    # it is located within ThirdParty, or a central installation
-    # outside of ThirdParty and must be added to the lib-path.
-
-    ending="${GPERFTOOLS_ARCH_PATH##*-}"
-    if [ "$ending" != none -a "$ending" != system ]
-    then
-        _foamAddLib     $GPERFTOOLS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-        _foamAddPath    $GPERFTOOLS_ARCH_PATH/bin
-    fi
-
-    unset gperftools_version ending
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/metis	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/metis	2017-07-11 10:10:57.483830647 +0200
@@ -54,8 +54,10 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$METIS_VERSION
+METIS_VERSION=metis-$EBVERSIONMETIS
+export METIS_ARCH_PATH=$EBROOTMETIS
+
+[ -d "$METIS_ARCH_PATH" ] || METIS_ARCH_PATH=$METIS_ROOT
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/mpi	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/mpi	2017-07-05 12:52:31.803773021 +0200
@@ -85,6 +85,10 @@
     _foamEtc config.sh/mpi-user                 # <- Adjustments
     ;;
 
+EASYBUILDMPI)  ##EasyBuild##
+    export FOAM_MPI=mpi
+    ;;
+
 SYSTEMMPI)
     export FOAM_MPI=mpi-system
     _foamEtc config.sh/mpi-system               # <- Adjustments (optional)
--- OpenFOAM-v1706.orig/etc/config.sh/paraview	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/paraview	2017-07-05 12:53:31.971688701 +0200
@@ -55,7 +55,7 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-ParaView_VERSION=5.4.0
+export ParaView_VERSION=$EBVERSIONPARAVIEW
 ParaView_MAJOR=detect                   # Automatically determine major version
 cmake_version=cmake-system
 
@@ -77,14 +77,6 @@
     ) && LD_LIBRARY_PATH="$cleaned"
 fi
 
-# ThirdParty cmake
-cmake=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cmake_version
-if [ -r $cmake/bin/cmake ]
-then
-    # _foamAddPath not available when foamPV function is used
-    PATH=$cmake/bin:$PATH
-fi
-
 # Evaluate command-line parameters for ParaView
 _foamParaviewEval()
 {
@@ -125,7 +117,8 @@
     pvName=ParaView-$ParaView_VERSION
     pvMajor=paraview-$ParaView_MAJOR
 
-    export ParaView_DIR=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$pvName
+    ##Easybuild## settings
+    export ParaView_DIR=$EBROOTPARAVIEW
 
     # Set paths if binaries are present
     if [ -r $ParaView_DIR ]
@@ -133,21 +126,10 @@
         pvLibDir=$ParaView_DIR/lib/$pvMajor
         pvPython=$ParaView_DIR/Utilities/VTKPythonWrapping
 
-        export PATH=$ParaView_DIR/bin:$PATH
+        ##EasyBuild## export PATH=$ParaView_DIR/bin:$PATH
         export ParaView_INCLUDE_DIR=$ParaView_DIR/include/$pvMajor
         export PV_PLUGIN_PATH=$FOAM_LIBBIN/$pvMajor
-        export LD_LIBRARY_PATH=$pvLibDir:$LD_LIBRARY_PATH
-
-        # Add in python libraries if required
-        if [ -r $pvPython ]
-        then
-            if [ "$PYTHONPATH" ]
-            then
-                export PYTHONPATH=$PYTHONPATH:$pvPython:$pvLibDir
-            else
-                export PYTHONPATH=$pvPython:$pvLibDir
-            fi
-        fi
+        ##EasyBuild## export LD_LIBRARY_PATH=$pvLibDir:$LD_LIBRARY_PATH
 
         if [ "$FOAM_VERBOSE" -a "$PS1" ]
         then
--- OpenFOAM-v1706.orig/etc/config.sh/scotch	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/scotch	2017-07-11 09:39:08.844419885 +0200
@@ -54,8 +54,10 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-SCOTCH_VERSION=scotch_6.0.3
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
+export SCOTCH_VERSION=scotch_$EBVERSIONSCOTCH
+export SCOTCH_ARCH_PATH=$EBROOTSCOTCH
+
+[ -d "$SCOTCH_ARCH_PATH" ] || SCOTCH_ARCH_PATH=$SCOTCH_ROOT
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
--- OpenFOAM-v1706.orig/etc/config.sh/settings	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/settings	2017-07-05 13:47:24.433587880 +0200
@@ -137,6 +137,13 @@
     ;;
 esac
 
+##Easybuild## compiler settings
+export WM_CC=$CC
+export WM_CXX=$CXX
+export WM_CFLAGS=$CFLAGS
+export WM_CXXFLAGS=$CXXFLAGS
+export WM_LDFLAGS=$LDFLAGS
+
 
 #------------------------------------------------------------------------------
 
--- OpenFOAM-v1706.orig/etc/config.sh/vtk	2017-06-30 10:50:14.000000000 +0200
+++ OpenFOAM-v1706/etc/config.sh/vtk	2017-07-05 12:54:06.355640418 +0200
@@ -43,40 +43,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-vtk_version=VTK-8.1.0
-mesa_version=mesa-17.1.1
-
-export VTK_DIR=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$vtk_version
-export MESA_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$mesa_version
+export VTK_DIR=$EBROOTVTK
+export MESA_ARCH_PATH=$EBROOTMESA
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ "$FOAM_VERBOSE" -a "$PS1" ]
-then
-    echo "Using vtk  ($vtk_version)  ->  $VTK_DIR" 1>&2
-    echo "Using mesa ($mesa_version)  ->  $MESA_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLib > /dev/null 2>&1  # normal sourcing
-then
-
-    # Set paths if binaries are present
-    if [ -r $VTK_DIR ]
-    then
-        # Uses lib/ and not lib64/
-        _foamAddLib $VTK_DIR/lib
-    else
-        unset VTK_DIR
-    fi
-
-    if [ -r $MESA_ARCH_PATH ]
-    then
-        _foamAddLib $MESA_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-    else
-        unset MESA_ARCH_PATH
-    fi
-
-    unset vtk_version mesa_version
-fi
-
-#------------------------------------------------------------------------------
